packages/contracts/contracts/IExecutorSelectionStrategy.sol

```diff
@@ -1,10 +1,59 @@
 // SPDX-License-Identifier: MIT
 pragma solidity ^0.8.9;
 
+import { ChugSplashRegistry } from "./ChugSplashRegistry.sol";
+
 /**
  * @notice Interface for the Executor Selection Strategy (ESS).
  */
 interface IExecutorSelectionStrategy {
+    /**
+     * @notice Emitted when an upgrade is claimed by an executor.
+     *
+     * @param project      Address of the ChugSplashManager that mananges the project.
+     * @param bundleId     ID of the bundle that was claimed.
+     * @param executor     Address of the executor that claimed the bundle ID for the project.
+     * @param prevExecutor Address of the executor that previously claimed the bundle ID for the
+     *                     project. This is address(0) if there was no previous executor.
+     */
+    event UpgradeClaimed(
+        address indexed project,
+        bytes32 indexed bundleId,
+        address indexed executor,
+        address prevExecutor
+    );
+
+    /**
+     * @notice Emitted when an executor is refunded the bond that they originally posted to claim an
+     *         upgrade.
+     *
+     * @param project  Address of the ChugSplashManager that mananges the project.
+     * @param bundleId ID of the bundle that was claimed.
+     * @param executor Address of the executor that posted the bond.
+     */
+    event ExecutorBondReturned(
+        address indexed project,
+        bytes32 indexed bundleId,
+        address indexed executor
+    );
+
+    /**
+     * @notice Allows an executor to claim the sole right to execute actions for an upgrade.
+     *
+     * @param _project  Address of the ChugSplashManager for the project being claimed.
+     * @param _bundleId ID of the bundle being claimed.
+     */
+    function claim(address _project, bytes32 _bundleId) external payable;
```

Comment:
I would make these functions specific to the `SimpleLockESS` instead of adding them to the `IExecutorSelectionStrategy`, since these aren't going to be necessary for every ESS.

Reply:
We use `claim` in the executor script [here](https://github.com/smartcontracts/chugsplash/blob/744545dfcfa92de0955b5cd811e87d815ad6486b/packages/executor/src/index.ts#L151), `getSelectedExecutor` in the ChugSplashManager [here](https://github.com/smartcontracts/chugsplash/blob/82af0f4eaacc7f7d61553c19f31fdbd6eb57f395/packages/contracts/contracts/ChugSplashManager.sol#L312), and `returnExecutorBond` is added to the ChugSplashManager in #50 [here](https://github.com/smartcontracts/chugsplash/blob/82af0f4eaacc7f7d61553c19f31fdbd6eb57f395/packages/contracts/contracts/ChugSplashManager.sol#L393) and [here](https://github.com/smartcontracts/chugsplash/blob/82af0f4eaacc7f7d61553c19f31fdbd6eb57f395/packages/contracts/contracts/ChugSplashManager.sol#L416). I think these functions should be kept in the interface unless we remove them from the Manager contract / executor script.

Reply:
I need to think about this a little more. I'm considering whether it might be worth scrapping the ESS/EPS adapter system for the alpha release. If we're making ChugSplash upgradeable, we could always add it back in later.

Reply:
Unfortunately `SimpleLock` does mean a lot of value goes to miners in a PGA. Maybe not worth worrying about for the alpha release, but we may need a simple staking mechanism eventually.

Reply:
Simple staking mechanism:
- Post at least 1 ETH and you get to be an executor
- Chance of being selected to execute a job depends on the % of the ETH pool that staked, time-weighted over the last month or something
- Slots for starting an upgrade are short (on the order of a few minutes) so that an upgrade will start within X amount of time on average assuming that Y% of the staking supply is honest
- Once you start you have Z blocks to finish where Z depends on the number of actions within the upgrade or something
- Since some upgrades might not be executable (let's say the thing returned by the IPFS link is just garbage data), you can't punish people for refusing to execute the upgrade (would make a really cool usecase for optimistic fault proofs down the line though!)

Reply:
Sounds like a good idea to add a staking mechanism like that in the future. Should we scrap the ESS/EPS system for the alpha release or leave it as is?

Reply:
Let's scrap both EPS and ESS in favor of simple lock for executor selection and flat fee (something like 10% above gas usage) for executor payment by default for the Alpha release.

Reply:
Sounds good, removed the ESS logic and replaced it with an `ExecutorSelector` contract.